cleanlab/datalab/datalab.py

```diff
@@ -0,0 +1,632 @@
+        pred_probs: Optional[np.ndarray] = None,
+        features: Optional[npt.NDArray] = None,
+        knn_graph: Optional[csr_matrix] = None,
+        model=None,  # sklearn.Estimator compatible object  # noqa: F821
+        issue_types: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Checks the dataset for all sorts of common issues in real-world data (in both labels and feature values).
+
+        You can use Datalab to find issues in your data, utilizing *any* model you have already trained.
+        This method only interacts with your model via its predictions or embeddings (and other functions thereof).
+        The more of these inputs you provide, the more types of issues Datalab can detect in your dataset/labels.
+        If you provide a subset of these inputs, Datalab will output what insights it can based on the limited information from your model.
+
+        Note
+        ----
+        The issues are saved in the ``self.issues`` attribute, but are not returned.
+
+        Parameters
+        ----------
+        pred_probs :
+            Out-of-sample predicted class probabilities made by the model for every example in the dataset.
+            To best detect label issues, provide this input obtained from the most accurate model you can produce.
+
+        features :
+            Feature embeddings (vector representations) of every example in the dataset.
+
+        knn_graph :
+            Sparse matrix representing similarities between examples in the dataset in a K nearest neighbor graph.
+            If both `knn_graph` and `features` are provided, the `knn_graph` will take precendence.
+            If `knn_graph` is not provided, it is constructed based on the provided `features`.
+            If neither `knn_graph` nor `features` are provided, certain issue types like (near) duplicates will not be considered.
+
+        model :
+            sklearn-compatible model trained via cross-validation to compute out-of-sample
+            predicted probabilities for each example. Only considered if `pred_probs` was not provided.
+
+            WARNING
+            -------
+            This is not yet implemented.
+            
+        issue_types :
+            Collection specifying which types of issues to consider in audit and any non-default parameter settings to use.
+            If unspecified, a default set of issue types and recommended parameter settings is considered.
+            
+            This is a dictionary of dictionaries, where the keys are the issue types of interest 
+            and the values are dictionaries of parameter values that control how each type of issue is detected.
+            More specifically, the values are constructor keyword arguments passed to the corresponding ``IssueManager``,
+            which is responsible for detecting the particular issue type.
+            
+            .. seealso::
+                IssueManager
+            
+            Example
```

Comment:
TODO: First provide a simpler example just showing how to only run a check for say label issues only (with default parameter-settings for this check)

Reply:
Added:

https://github.com/cleanlab/cleanlab/pull/614/commits/31d86f4001578866ba1f2d9d1142d7ba8dcefccf#diff-e126cf653842496e91462249d07fa417cca997b74bc9924d4c80d9c715c11ea0R497-R508